scrape/scrape_test.go

```diff
@@ -4757,70 +4754,34 @@ metric: <
 				simpleStorage := teststorage.New(t)
 				defer simpleStorage.Close()
 
-				config := &config.ScrapeConfig{
-					JobName:                        "test",
-					SampleLimit:                    100,
-					Scheme:                         "http",
-					ScrapeInterval:                 model.Duration(50 * time.Millisecond),
-					ScrapeTimeout:                  model.Duration(49 * time.Millisecond),
-					AlwaysScrapeClassicHistograms:  tc.alwaysScrapeClassicHistograms,
-					ConvertClassicHistogramsToNHCB: tc.convertClassicHistToNHCB,
-					MetricNameValidationScheme:     model.UTF8Validation,
-					MetricNameEscapingScheme:       model.AllowUTF8,
-				}
-
-				scrapeCount := 0
-				scraped := make(chan bool)
-
-				ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
-					if metricsText.contentType != "" {
-						w.Header().Set("Content-Type", `application/vnd.google.protobuf; proto=io.prometheus.client.MetricFamily; encoding=delimited`)
-						for _, text := range metricsText.text {
-							buf := &bytes.Buffer{}
-							// In case of protobuf, we have to create the binary representation.
-							pb := &dto.MetricFamily{}
-							// From text to proto message.
-							require.NoError(t, proto.UnmarshalText(text, pb))
-							// From proto message to binary protobuf.
-							protoBuf, err := proto.Marshal(pb)
-							require.NoError(t, err)
-
-							// Write first length, then binary protobuf.
-							varintBuf := binary.AppendUvarint(nil, uint64(len(protoBuf)))
-							buf.Write(varintBuf)
-							buf.Write(protoBuf)
-							w.Write(buf.Bytes())
-						}
-					} else {
-						for _, text := range metricsText.text {
-							fmt.Fprint(w, text)
-						}
-					}
-					scrapeCount++
-					if scrapeCount > 2 {
-						close(scraped)
+				sl := newBasicScrapeLoop(t, context.Background(), nil, func(ctx context.Context) storage.Appender { return simpleStorage.Appender(ctx) }, 0)
+				sl.alwaysScrapeClassicHist = tc.alwaysScrapeClassicHistograms
+				sl.convertClassicHistToNHCB = tc.convertClassicHistToNHCB
+				sl.enableNativeHistogramIngestion = true
+				app := simpleStorage.Appender(context.Background())
+
+				var content []byte
+				contentType := metricsText.contentType
+				switch contentType {
+				case "application/vnd.google.protobuf":
```

Comment:
Maybe something for a different PR but my first reaction is this should be a constant.

Apparently it is written out this way in 17 places already in this repo.

Reply:
I'll open a PR for that after this (sometime later today)